kernel/src/ipc.rs

```diff
@@ -233,54 +323,35 @@ impl<const NUM_PROCS: usize> LegacyDriver for IPC<NUM_PROCS> {
         &self,
         appid: AppId,
         target_id: usize,
-        slice: Option<AppSlice<SharedReadWrite, u8>>,
-    ) -> ReturnCode {
+        mut slice: ReadWriteAppSlice,
+    ) -> Result<ReadWriteAppSlice, (ReadWriteAppSlice, ErrorCode)> {
         if target_id == 0 {
```

Comment:
I am guessing this is the original reason that the target id was app_id + 1? Now that allow does not use target_id == 0, it seems we could go back to target_id = app_id. But maybe there is a benefit in keeping them separate in that it forces apps not to assume they are equal, which is something the kernel does not guarantee

Reply:
Right, I hadn't thought through all the implications so just left it as is for now. After porting libtock-c, I'll see if we can just elide this transformation. I don't think there is a significant benefit to obscuring this intentionally from apps if it doesn't serve another purpose (the semantics should still not promise anything about the meaning of the numbers though)